app/javascript/controllers/data_exports/metadata_selection_controller.js

```diff
@@ -0,0 +1,69 @@
+import { Controller } from "@hotwired/stimulus";
+import { createHiddenInput } from "utilities/form";
+
+export default class extends Controller {
+
+  static targets = ["field", "submitBtn"];
+
+  static values = {
+    selectedList: {
+      type: String,
+    },
+    availableList: {
+      type: String,
+    }
+  };
+
+  connect() {
+    this.availableList = document.getElementById(this.availableListValue)
+    this.selectedList = document.getElementById(this.selectedListValue)
+    this.fullListItems = this.availableList.querySelectorAll("li")
+    this.selectedList.addEventListener("mouseover", () => { this.#checkSubmitState() })
+  }
+
+  addAll() {
+    for (const item of this.fullListItems) {
+      this.selectedList.append(item)
+    }
+
+    this.#setSubmitButtonDisableState(false)
```

Comment:
When a user adds all items to the selected list, we should disable the link for `Add All` link as all the fields have been added. Same applies if there is no items in the selected list, the `Remove All` link should be disabled